### PR TITLE
Introduce command criterion value __focused__

### PR DESCRIPTION
**Pull request by Airblader**
relates to #1770

### CODE REVIEW

File: testcases/t/251-command-criteria-focused.t [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+is(@{get_ws($ws)->{nodes}}, 2, 'sanity check: workspace contains two windows');
+
+cmd '[title=__focused__] move to workspace trash';
+is(@{get_ws($ws)->{nodes}}, 0, '__focused__ works for title');
+
+###############################################################################
+# 4: Test __focused__ for window role.
+###############################################################################
+
+# NOTE X11::XCB currently doesn't support setting the window role.
```
Comment (Airblader) [inline testcases/t/251-command-criteria-focused.t]: @stapelberg Is there a good way to set the window role?

Comment (stapelberg) [inline testcases/t/251-command-criteria-focused.t]: https://github.com/i3/i3/blob/b68a400abfeb9f17901497eb005862853656287c/testcases/t/214-layout-restore-criteria.t#L93 currently is the best way.

File: docs/userguide [hunk 2]
```diff
@@ -1604,11 +1604,17 @@ for_window [class="^evil-app$"] floating enable, move container to workspace 4
 The criteria which are currently implemented are:
 
 class::
-	Compares the window class (the second part of WM_CLASS)
+        Compares the window class (the second part of WM_CLASS). Use the
+        special value +__focused__+ to match all windows having the same window
+        class as the currently focused window.
```
Comment (stapelberg) [inline docs/userguide]: Here and below, it looks like you’re changing indentation. That doesn’t seem to be intentional?

Comment (Airblader) [inline docs/userguide]: It wasn't intentional (I haven't noticed), but I formatted it with vim (`gq`). It's now 8 spaces, before it was a tab. What's your preference? In the code we use spaces.

File: testcases/t/251-command-criteria-focused.t (hunk 1 above)
Comment (Airblader) [inline testcases/t/251-command-criteria-focused.t]: Ah, thanks. I'll put that in.

File: docs/userguide (hunk 2 above)
Comment (stapelberg) [inline docs/userguide]: I’d prefer it if you could keep it consistent with how the file is currently formatted.

**Comment by Airblader** [comment]
@stapelberg Done. The format change in there now should be correct because it was using spaces before.

**Comment by stapelberg** [comment]
In its current form, the test will pass when __focused__ just matches any window. Can you open at least one more window with a different class to make sure that’s not the case, please?

**Comment by Airblader** [comment]
@stapelberg Done.

File: docs/userguide [hunk 3]
```diff
@@ -1604,25 +1604,35 @@ for_window [class="^evil-app$"] floating enable, move container to workspace 4
...
+	Compares the window role (WM_WINDOW_ROLE). Use the special value
+	+__focused__+ to match all windows having the same window role as the
+	currently focused window.
 window_type::
-        Compare the window type (_NET_WM_WINDOW_TYPE). Possible values are
-        +normal+, +dialog+, +utility+, +toolbar+, +splash+, +menu+, +dropdown_menu+,
-        +popup_menu+ and +toolti+.
+	Compare the window type (_NET_WM_WINDOW_TYPE). Possible values are
+	+normal+, +dialog+, +utility+, +toolbar+, +splash+, +menu+, +dropdown_menu+,
+	+popup_menu+ and +toolti+.
```
Comment (Airblader) [inline docs/userguide]: There's a typo here. I'll fix this in a separate PR once this is merged.

**Pull request merged**